client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java

```diff
@@ -22,35 +22,88 @@
-    public String server;
+    private static final String SERVER_DEFAULT = "localhost:8080";
+    private static final String SOURCE_DEFAULT = "trino-cli";
+
+    // some entries have different semantics, but this mapping is only used to disallow setting both as a CLI option and URL parameter
+    static final Map<String, PropertyName> optionsToProperties = ImmutableMap.<String, PropertyName>builder()
+            .put("--krb5-service-principal-pattern", PropertyName.KERBEROS_SERVICE_PRINCIPAL_PATTERN)
+            .put("--krb5-config-path", PropertyName.KERBEROS_CONFIG_PATH)
+            .put("--krb5-keytab-path", PropertyName.KERBEROS_KEYTAB_PATH)
+            .put("--krb5-credential-cache-path", PropertyName.KERBEROS_CREDENTIAL_CACHE_PATH)
+            .put("--krb5-principal", PropertyName.KERBEROS_PRINCIPAL)
+            .put("--krb5-disable-remote-service-hostname-canonicalization", PropertyName.KERBEROS_USE_CANONICAL_HOSTNAME)
+            .put("--keystore-path", PropertyName.SSL_KEY_STORE_PATH)
+            .put("--keystore-password", PropertyName.SSL_KEY_STORE_PASSWORD)
+            .put("--keystore-type", PropertyName.SSL_KEY_STORE_TYPE)
+            .put("--truststore-path", PropertyName.SSL_TRUST_STORE_PATH)
+            .put("--truststore-password", PropertyName.SSL_TRUST_STORE_PASSWORD)
+            .put("--truststore-type", PropertyName.SSL_TRUST_STORE_TYPE)
+            .put("--use-system-truststore", PropertyName.SSL_USE_SYSTEM_TRUST_STORE)
+            .put("--insecure", PropertyName.SSL_VERIFICATION)
+            .put("--access-token", PropertyName.ACCESS_TOKEN)
+            .put("--user", PropertyName.USER)
+            // password should never be allowed in the URL
+            .put("--external-authentication", PropertyName.EXTERNAL_AUTHENTICATION)
+            .put("--external-authentication-redirect-handler", PropertyName.EXTERNAL_AUTHENTICATION_REDIRECT_HANDLERS)
+            .put("--source", PropertyName.SOURCE)
+            .put("--client-info", PropertyName.CLIENT_INFO)
+            .put("--client-tags", PropertyName.CLIENT_TAGS)
+            .put("--trace-token", PropertyName.TRACE_TOKEN)
+            .put("--session-user", PropertyName.SESSION_USER)
+            .put("--session", PropertyName.SESSION_PROPERTIES)
+            .put("--extra-credential", PropertyName.EXTRA_CREDENTIALS)
+            .put("--socks-proxy", PropertyName.SOCKS_PROXY)
+            .put("--http-proxy", PropertyName.HTTP_PROXY)
+            .put("--disable-compression", PropertyName.DISABLE_COMPRESSION)
+            // these are not properties, but defining them here will help to map the property from an exception back to an option
+            .put("--catalog", PropertyName.CATALOG)
+            .put("--schema", PropertyName.SCHEMA)
+            .buildOrThrow();
+
+    @Parameters(paramLabel = "URL", description = "Trino server URL", arity = "0..1")
```

Comment:
let's call the label `Connection URI` and `Trino connection URI`

Reply:
But it's not a URI, it has to be a valid location.

Reply:
Also, why the longer version? It shows up in help as:

```

trino [-h] [--debug] ... [URL]
      [URL]                  Trino server URL
```